src/repr/src/adt/range.rs

```diff
@@ -173,6 +186,56 @@ impl<D> Range<D> {
         }
     }
 
+    /// Get the flag bits appropriate to use in our internal (i.e. row) encoding
+    /// of range values.
+    ///
+    /// Note that this differs from the flags appropriate to encode with
```

Comment:
This doesn't seem that bad since it only means we do a bit of munging when serializing over pgwire?

Reply:
Yeah, and we would ostensibly need to do the inverse munging when deserializing. This worries me slightly only because the bad case (using the wrong bits) can be a totally opaque error––but agreed that it, generally, doesn't seem that bad.